internal/yggdrasil/yggdrasil.go

```diff
@@ -310,33 +317,66 @@ func (h *Handler) updateDeploymentStatuses(oldDeployments []v1alpha1.Deployment,
 	return deployments
 }
 
-func (h *Handler) toWorkloadList(logger logr.Logger, deployments []v1alpha1.EdgeDeployment) models.WorkloadList {
+func (h *Handler) toWorkloadList(ctx context.Context, logger logr.Logger, deployments []v1alpha1.EdgeDeployment, device *v1alpha1.EdgeDevice) (models.WorkloadList, error) {
 	list := models.WorkloadList{}
 	for _, deployment := range deployments {
 		if deployment.DeletionTimestamp != nil {
 			continue
 		}
-		podSpec, err := yaml.Marshal(deployment.Spec.Pod.Spec)
+		spec := deployment.Spec
+		podSpec, err := yaml.Marshal(spec.Pod.Spec)
 		if err != nil {
 			logger.Error(err, "cannot marshal pod specification", "deployment name", deployment.Name)
 			continue
 		}
 		var data *models.DataConfiguration
-		if deployment.Spec.Data != nil && len(deployment.Spec.Data.Paths) > 0 {
+		if spec.Data != nil && len(spec.Data.Paths) > 0 {
 			var paths []*models.DataPath
-			for _, path := range deployment.Spec.Data.Paths {
+			for _, path := range spec.Data.Paths {
 				paths = append(paths, &models.DataPath{Source: path.Source, Target: path.Target})
 			}
 			data = &models.DataConfiguration{Paths: paths}
 		}
+
 		workload := models.Workload{
 			Name:          deployment.Name,
 			Specification: string(podSpec),
 			Data:          data,
 		}
+		authFile, err := h.getAuthFile(ctx, spec.ImageRepositories, deployment.Namespace)
+		if err != nil {
+			h.recorder.Eventf(device, corev1.EventTypeWarning, "Misconfiguration",
+				"Auth file secret %s used by deployment %s/%s is missing", spec.ImageRepositories.AuthFileSecret.Name, deployment.Namespace, deployment.Name,
+			)
+			logger.Error(err, "Can't get auth file")
+			return nil, err
+		}
+		if authFile != "" {
+			workload.ImageRepositories = &models.ImageRepositories{
+				AuthFile: authFile,
+			}
+		}
 		list = append(list, &workload)
 	}
-	return list
+	return list, nil
+}
+
+func (h *Handler) getAuthFile(ctx context.Context, imageRepositories *v1alpha1.ImageRepositoriesConfiguration, defaultNamespace string) (string, error) {
+	if imageRepositories != nil {
+		if secretRef := imageRepositories.AuthFileSecret; secretRef != nil {
+			namespace := secretRef.Namespace
+			if secretRef.Namespace == "" {
+				namespace = defaultNamespace
```

Comment:
Shouldn't we prefer the namespace of the edge-device if it wasn't explicitly provided ?

Reply:
`defaultNamespace` here comes from the parameter, which in turn is set with `deployment.Namespace` value, which is equal to device namespace.

Reply:
Right...I somehow assumed it is a constant defined as 'default' and missed the fact it was provided as a parameter :-)